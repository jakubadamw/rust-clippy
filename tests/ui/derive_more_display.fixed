//@run-rustfix
#![feature(rustc_private)]
#![allow(unused)]
#![warn(clippy::derive_more_display)]

extern crate derive_more;

mod x {
    

    struct X {}

    #[derive(derive_more::Display)]
#[display("{:?}, world {}", "self.first", "self.second.0")]
struct Type {
        first: usize,
        second: (usize, usize),
    }
}

fn main() {}
