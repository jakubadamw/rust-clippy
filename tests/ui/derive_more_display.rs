//@run-rustfix
#![feature(rustc_private)]
#![allow(unused)]
#![warn(clippy::derive_more_display)]

extern crate derive_more;

mod x {
    impl std::fmt::Display for Type {
        fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
            write!(f, "{:?}, world {}", self.first, self.second.0)
        }
    }

    struct X {}

    struct Type {
        first: usize,
        second: (usize, usize),
    }
}

fn main() {}
