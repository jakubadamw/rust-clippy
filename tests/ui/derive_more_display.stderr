error: this `impl` can be derived
  --> $DIR/derive_more_display.rs:9:5
   |
LL | /     impl std::fmt::Display for Type {
LL | |         fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
LL | |             write!(f, "{:?}, world {}", self.first, self.second.0)
LL | |         }
LL | |     }
   | |_____^
   |
   = note: `-D clippy::derive-more-display` implied by `-D warnings`
   = help: remove the manual implementation...
help: ...and instead derive it...
   |
LL ~     #[derive(derive_more::Display)]
LL + #[display("{:?}, world {}", "self.first", "self.second.0")]
LL |     struct Type {
   |

error: aborting due to previous error

